packages/payment-detection/src/thegraph/index.ts

```diff
@@ -9,24 +10,29 @@ const BASE_URL = `https://api.thegraph.com`;
 /**
  * A GraphQL client to query Request's subgraph.
  */
-export type TheGraphClient = ReturnType<typeof getSdk>;
+export type TheGraphClient<
+  TGraphClientVariant extends 'near' | null = null
+> = TGraphClientVariant extends 'near' ? ReturnType<typeof getSdkNear> : ReturnType<typeof getSdk>;
 export type TheGraphClientOptions = {
   baseUrl?: string;
   timeout?: number;
 };
 
-export const getTheGraphClient = (
+export const getTheGraphClient = <TGraphClientVariant extends 'near' | null = null>(
   network: string,
   options?: TheGraphClientOptions,
-): TheGraphClient => {
+) => {
   const baseUrl = options?.baseUrl || network === 'private' ? 'http://localhost:8000' : BASE_URL;
   // Note: it is also possible to use the IPFS hash of the subgraph
   //  eg. /subgraphs/id/QmcCaSkefrmhe4xQj6Y6BBbHiFkbrn6UGDEBUWER7nt399
   //  which is a better security but would require an update of the
   //  library each time the subgraph is updated, which isn't ideal
   //  for early testing.
   const url = `${baseUrl}/subgraphs/name/requestnetwork/request-payments-${network}`;
-  return getSdk(new GraphQLClient(url, options));
+  if (network === 'near') {
+    return getSdkNear(new GraphQLClient(url, options)) as TheGraphClient<TGraphClientVariant>;
```

Comment:
Is the cast necessary ? Specifying the return type of the method doesn't work here ?

Reply:
Pretty sure yes. Another option would be to not make the method generic, and instead have 2 types `TheGraphClient` and `TheGraphNearClient`, and two methods `getTheGraphClient` and `getTheGraphNearClient`.

I think it actually makes sense because the genericity of this method is not used and since the types are quite different, I don't see how it could factorize anything.

I'll remove it.

Reply:
That's why I've thought it was easier for me to have a separated file for superfluid subgraph methods which can be imported in Superfluid info retriever instead of the existing one.